### PR TITLE
Byte level care

### PR DESCRIPTION
**Pull request by szepeviktor**
@mnot This PR goes into "321" branch.

These commits are the result of [the Integrity workflow](https://github.com/szepeviktor/byte-level-care/blob/master/.github/workflows/reusable-integrity.yml). It cares about every byte.
Please see individual commits.

Review without whitespace changes: https://github.com/mnot/redbot/pull/331/files?diff=unified&w=1

### CODE REVIEW

**Comment by szepeviktor** [comment]
I run the above workflow on each commit to keep my bytes safe.

**Comment by szepeviktor** [comment]
I am not about `redbot/assets/red_style.css` as it is a generated file but not minimized.

**Comment by mnot** [comment]
I suspect css indentation should be set to `2`, just like scss. I'll do that.

File: redbot/assets/red_style.css
Comment (mnot) [inline redbot/assets/red_style.css:1]: Should be cut back to two space indent now.

Comment (szepeviktor) [reply redbot/assets/red_style.css:1]: Done.

**Comment by mnot** [comment]
Two things:

1. I use double blank lines above Markdown section headings and between classes in Python to give visual space, so I'd prefer to leave those alone. Also, `Makefile.venv` is from an external source, so it shouldn't be changed.
2. What's the motivation for not archiving development artefacts? Being able to eg download a zip from GitHub and run tests is useful.

Otherwise looks great, thanks!

**Comment by szepeviktor** [comment]
> What's the motivation for not archiving development artefacts? Being able to eg download a zip from GitHub and run tests is useful.

I though people using **git**hub are using `git` to get source code ...

**Comment by szepeviktor** [comment]
> 1. I use double blank lines above Markdown section headings and between classes in Python

Done 🍏

**Comment by mnot** [comment]
Thanks.

Re: .gitattributes - hm. I don't see archiving as the way to get to production, but if you do, you're likely to want to use something like `make server` to get the environment set up. Also the archive can be useful for doing things like downloading a zip to check a difference in a PR. Do these files do any harm?

**Comment by szepeviktor** [comment]
> downloading a zip to check a difference in a PR.

This activity also resembles someone not familiar with git.
ZIP archives (`git archive HEAD`) are usually used by package managers to get package.
It may not be true for redbot.

A second thought: redbot may not be a library but an application ...

**Comment by mnot** [comment]
> A second thought: redbot may not be a library but an application ...

Exactly...

**Comment by szepeviktor** [comment]
Deleted!

**Comment by mnot** [comment]
Thanks!

**Comment by szepeviktor** [comment]
Glad to meet your bytes.

**Pull request merged**